### PR TITLE
SC remove pinned env

### PR DESCRIPTION
**Pull request by velzevur**
PT [SC: Remove pinned env from FP and offchain tx](https://www.pivotaltracker.com/story/show/165593706)

This is `aeternity` node's part of the ticket

### CODE REVIEW

File: apps/aechannel/src/aesc_force_progress_tx.erl [hunk 1]
```diff
@@ -74,53 +58,13 @@
           offchain_trees:: aec_trees:trees(),
           ttl           :: aetx:tx_ttl(),
           fee           :: non_neg_integer(),
-          nonce         :: non_neg_integer(),
-          block_hash    :: aesc_pinned_block:hash()
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_force_progress_tx.erl]: Is there a released version of the software with such record? I understand yes. So this breaks DB backward compatibility, that may be ok with 3.* though we need to decide this.

Comment (velzevur) [reply apps/aechannel/src/aesc_force_progress_tx.erl]: Those records are allowed only from Fortuna on, so assumption is no such records on main net

File: config/swagger.yaml [hunk 2]
```diff
@@ -2638,9 +2638,6 @@ definitions:
       state_hash:
         $ref: '#/definitions/EncodedHash'
         description: 'Channel''s next state_hash'
-      block_hash:
```
Comment (lucafavatella) [inline config/swagger.yaml]: As this was re feature introduced for Fortuna though never used, this shall not impact users. Though I suggest adding release note line mentioning removal of this.

File: apps/aechannel/src/aesc_force_progress_tx.erl [hunk 3]
```diff
@@ -391,21 +358,14 @@ state_hash(#channel_force_progress_tx{state_hash = StateHash}) ->
...
 -spec version(tx()) -> non_neg_integer().
-version(#channel_force_progress_tx{block_hash = BlockHash}) ->
-    case BlockHash =:= aesc_pinned_block:no_hash() of
-        true ->
-            ?INITIAL_VSN;
-        false ->
-            ?PINNED_BLOCK_VSN
-    end.
+version(_) ->
+    ?LATEST_VSN.
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_force_progress_tx.erl]: It seems to me we rather want to keep INITIAL_VSN i.e. 1, and optionally skip 2 in future?

Comment (lucafavatella) [reply apps/aechannel/src/aesc_force_progress_tx.erl]: I need to understand better if this is in serialization or DB though.

File: apps/aechannel/src/aesc_force_progress_tx.erl [hunk 4]
```diff
@@ -74,23 +62,35 @@
...
+         }).
 %%%===================================================================
 %%% Conversion of old db format
 
--spec from_db_format(tx()) -> tx().
+ -spec from_db_format(tx()) -> tx().
 from_db_format(#channel_force_progress_tx{} = Tx) ->
     Tx;
 from_db_format(Tuple) ->
     case setelement(1, Tuple, old_fp_db_record) of
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_force_progress_tx.erl]: Bad atom?

File: apps/aechannel/src/aesc_force_progress_tx.erl (hunk 3 above)
Comment (lucafavatella) [reply apps/aechannel/src/aesc_force_progress_tx.erl]: As I understand from https://github.com/aeternity/protocol/pull/341/files
using 3 is unneeded. A simple revert to 1 is ok?

Also - could this version 2 anywhere got to the DB for supported runtime i.e. Minerva? I expect no.

File: apps/aechannel/src/aesc_force_progress_tx.erl (hunk 4 above)
Comment (velzevur) [reply apps/aechannel/src/aesc_force_progress_tx.erl]: Thanks

File: config/swagger.yaml (hunk 2 above)
Comment (lucafavatella) [reply config/swagger.yaml]: Still suggested though not strictly needed.

File: apps/aechannel/src/aesc_offchain_tx.erl [hunk 5]
```diff
@@ -56,8 +54,7 @@
           channel_id         :: aeser_id:id(),
           updates            :: [aesc_offchain_update:update()],
           state_hash         :: binary(),
-          round              :: non_neg_integer(),
-          block_hash         :: aesc_pinned_block:hash()
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_offchain_tx.erl]: So this was not stored in DB as record.

File: apps/aechannel/src/aesc_force_progress_tx.erl [hunk 6]
```diff
@@ -74,23 +60,35 @@
...
-          nonce         :: non_neg_integer(),
-          block_hash    :: aesc_pinned_block:hash()
+          nonce         :: non_neg_integer()
          }).
 
 -opaque tx() :: #channel_force_progress_tx{}.
 
 -export_type([tx/0]).
 
+-record(v2_db_record, {
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_force_progress_tx.erl]:
```suggestion
%% Record introduced temporarily during Fortuna development, shipped in releases 2.4.0 and 2.5.0, and potentially stored in DB.
-record(v2_db_record, {	
```

File: apps/aechannel/src/aesc_offchain_tx.erl (hunk 5 above)
Comment (velzevur) [reply apps/aechannel/src/aesc_offchain_tx.erl]: It was always stored in a serialized format and never as a record

**Pull request merged**